### ISSUE TITLE
Basic feature missing

### ISSUE THREAD

**Issue by davidmoseler** [pytorch/pytorch#52762]
Pytorch is based on autograd. When I calculate a backward pass, for example, y = x**2, y.backward(), pytorch is missing the extremely basic feature to be able to get x from y, that is, to display where the gradient will be propagated to. Of course, internally pytorch needs to have this, inside the so called "computational graph", but it seems that developers forgot that not only pytorch internals, but users of pytorch need to be able to get the computational graph too. If y.backward() propagates gradients to the tensor x, I should be able to see to which tensor the gradients will be or are being propagated. If this method exists, it is not documented as far as I could find.

**Comment by anjali411** [comment]
@davidmoseler did you check out our autograd visualizer tool called `torchviz`? you can download it by running `pip install torchviz` and then running this would show the current computational graph.

```
import torch
import torchviz
x = torch.rand(5, 2, requires_grad=True)

y = x**2

torchviz.make_dot((y.sum(), x))
```

**Comment by davidmoseler** [comment]
Hi,

The idea would not be visualization, but manual traversing of the graph. Apparently, it can be traversed with the ".next_functions" method, but it doesn't have a documentation.

**Comment by albanD** [comment]
Hi,

This is not documented because we don't consider this to be a nice enough API. There are small gotchas that can be surprising to users when traversing the graph especially when view/inplace are involved.

Do you have a specific use case in mind?
